pkg/trimaran/targetloadpacking/README.md

```diff
@@ -12,16 +12,36 @@ This folder holds the `TargetLoadPacking` plugin implementation based on [Trimar
 - [ ] 👨 Stable (used in companies for production workloads)
 
 ## TargetLoadPacking Plugin
-`TargetLoadPacking` depends on [Load Watcher](https://github.com/paypal/load-watcher) service. Instructions to build and deploy load watcher can be found [here](https://github.com/paypal/load-watcher/blob/master/README.md).
-`watcherAddress` argument below must be setup for `TargetLoadPacking` to work.
+`TargetLoadPacking` depends on [Load Watcher](https://github.com/paypal/load-watcher).
+It uses `load-watcher` in two modes.
+1. Using `load-watcher` service as a metric endpoint.
+   You can run `load-watcher` service separately to provide real time node resource usage metrics for `TargetLoadPacking` to consume.
+   Instructions to build and deploy load watcher can be found [here](https://github.com/paypal/load-watcher/blob/master/README.md).
+   In this way, you just need to configure `watcherAddress: http://xxxx.svc.cluster.local` to your `load-watcher` service. You can 
+   also deploy `load-watcher` as a service in the same scheduler pod, 
+   following the tutorial [here](https://medium.com/paypal-engineering/real-load-aware-scheduling-in-kubernetes-with-trimaran-a8efe14d51e2).
+   
+2. Using `load-watcher` as a library to fetch metrics from other providers, such as Prometheus, SignalFx and Kubernetes metric server.
+   In this mode, you need to configure three parameters: `metricProviderType`, `metricProviderAddress` and `metricProviderToken` if authentication is needed.
+
+   By default, `metricProviderType` is ` KubernetesMetricsServer` if not set. Now it supports `KubernetesMetricsServer`, `Prometheus` and `SignalFx`.
+    - `metricProviderType: KubernetesMetricsServer` use `load-watcher` as a client library to retrieve metrics from Kubernetes metric
+      server directly.
```

Comment:
```suggestion
      server.
```
Technically it is indirect (metrics flow is provider -> watcher -> client) in both using it as a service and library.